### PR TITLE
[ubsan] fix some ubsan complains on vector and pointer

### PR DESCRIPTION
**Pull request by dataroaring**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exprs/expr_context.cpp [hunk 1]
```diff
@@ -116,7 +116,8 @@ Status ExprContext::clone(RuntimeState* state, ExprContext** new_ctx) {
     for (int i = 0; i < _fn_contexts.size(); ++i) {
         (*new_ctx)->_fn_contexts.push_back(_fn_contexts[i]->impl()->clone((*new_ctx)->_pool.get()));
     }
-    (*new_ctx)->_fn_contexts_ptr = &((*new_ctx)->_fn_contexts[0]);
+    (*new_ctx)->_fn_contexts_ptr = (*new_ctx)->_fn_contexts.size() == 0 ?
```
Comment (morningman) [inline be/src/exprs/expr_context.cpp:119]: Looks like no where to use this `_fn_contexts_ptr`. Can be deleted?

File: be/src/exprs/expr_context.h [hunk 2]
```diff
@@ -165,11 +165,6 @@ class ExprContext {
     /// and owned by this ExprContext.
     std::vector<FunctionContext*> _fn_contexts;
 
-    /// Array access to fn_contexts_. Used by ScalarFnCall's codegen'd compute function
-    /// to access the correct FunctionContext.
-    /// TODO: revisit this
-    FunctionContext** _fn_contexts_ptr;
```
Comment (zbtzbtzbt) [inline be/src/exprs/expr_context.h:171]: may i ask how to use this ubsan tool?

File: be/src/exprs/expr_context.cpp (hunk 1 above)
Comment (morningman) [reply be/src/exprs/expr_context.cpp:119]: You forgot to remove here.

File: be/src/exprs/expr_context.h (hunk 2 above)
Comment (dataroaring) [reply be/src/exprs/expr_context.h:171]: [p](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html)

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**